modules/stereo/src/matching.hpp

```diff
@@ -0,0 +1,364 @@
+                                    else
+                                    {
+                                        map[iw + j] = 0;
+                                    }
+                                }
+                                else
+                                {
+                                    if (width - j <= disparity)
+                                    {
+                                        p2 = Matching::symetricVInterpolation(c, iw + j, disparity + 1, lr,CV_VERTICAL_SEARCH);
+                                        map[iw + j] = (uint8_t)(p2* scallingFact);
+                                    }
+                                }
+                            }
+                            else
+                            {
+                                map[iw + j] = 0;
+                            }
+                        }
+                    }
+                }
+            };
+            //!median 1x9 paralelized filter
+            class Median1x9:public ParallelLoopBody
+            {
+            private:
+                uint8_t *harta;
+                uint8_t *mapModified;
+                int height, width;
+            public:
+                Median1x9(const Mat &hartaOriginala, Mat &map)
+                {
+                    harta = hartaOriginala.data;
+                    mapModified = map.data;
+                    height = hartaOriginala.rows;
+                    width = hartaOriginala.cols;
+                }
+                void operator()(const cv::Range &r) const{
+                    for (int m = r.start; m <= r.end; m++)
+                    {
+                        for (int n = 4; n < width - 4; ++n)
+                        {
+                            int k = 0;
+                            uint8_t window[9];
+                            for (int i = n - 4; i <= n + 4; ++i)
+                                window[k++] = harta[m * width + i];
+                            for (int j = 0; j < 5; ++j)
+                            {
+                                int min = j;
+                                for (int l = j + 1; l < 9; ++l)
+                                    if (window[l] < window[min])
+                                        min = l;
+                                const uint8_t temp = window[j];
+                                window[j] = window[min];
+                                window[min] = temp;
+                            }
+                            mapModified[m  * width + n] = window[4];
+                        }
+                    }
+                }
+            };
+            //!median 9x1 paralelized filter
+            class Median9x1:public ParallelLoopBody
+            {
+            private:
+                uint8_t *harta;
+                uint8_t *mapModified;
+                int height, width;
+            public:
+                Median9x1(const Mat &hartaOriginala, Mat &map)
```

Comment:
`hartaOriginala` is this English? =)

Anyways, median filter is a very generic function - it takes a Mat and returns it slightly modified. The physical meaning of the values in Mat can be arbitrary. So, there's no reason to use specific words, like "harta". It can as well be an image, not a "disparitate harta" =)

And of course, using both "harta" and "map" (in the same function signature, by the way!) doesn't seem like a buna idee =)